src/common/util/jsonPatcher.js

```diff
@@ -182,8 +199,167 @@ define(['common/util/canon'], function (CANON) {
         return result;
     }
 
+    function _endsWith(str, pattern) {
+        var d = str.length - pattern.length;
+        return d >= 0 && str.lastIndexOf(pattern) === d;
+    }
+
+    function _isOvr(path) {
+        return path.indexOf('/ovr/') === 0;
+    }
+
+    function _isRelid(path) {
+        return RANDOM.isValidRelid(path.substring(1));
+    }
+
+    function _isGmePath(path) {
+        var relIds = path.split('/'),
+            result = false,
+            i;
+
+        for (i = 1; i < relIds.length; i += 1) {
+            if (RANDOM.isValidRelid(relIds[i]) === false) {
```

Comment:
All path-pieces need to pass the check and at least one needs to pass it too - so I can return after one truthy check. However I agree that it is a bit unclear from this code.